src/helpers/report-helpers.ts

```diff
@@ -0,0 +1,122 @@
+import axios from 'axios'
+
+// eslint-disable-next-line import/no-cycle
+import { ReportsApiClientV20210630 } from '../api-clients/reports-api-client-v20210630'
+import { AMAZON_US_MARKETPLACE_ID } from './constants'
+import { sleep, tabDelimitedToArray } from './utils'
+
+interface ReportParameters {
+  startDate: string
+  endDate: string
+  marketplaceIds?: string[]
+}
+
+/**
+ * Allow other people to use the parsing function
+ */
+export { tabDelimitedToArray as parseAmazonReport }
+
+/**
+ * Fetch the latest parsed data of a report
+ */
+export async function getLatestReportHelper(
+  reportsClient: ReportsApiClientV20210630,
+  reportType: string,
+  parse = true,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): Promise<string | any> {
+  // Get a list of the latest report
+  const getReportResponse = await reportsClient.getReports({
+    reportTypes: [reportType],
+    processingStatuses: ['DONE'],
+    pageSize: 1,
+  })
+
+  // ensure we have a document ID
+  const reportDocumentId = getReportResponse.data.reports?.[0]?.reportDocumentId || ''
+  if (!reportDocumentId) {
+    throw new Error(`No report for ${reportType}`)
+  }
+
+  // fetch the report document
+  const documentResponse = await reportsClient.getReportDocument({
+    reportDocumentId,
+  })
+
+  // fetch the raw content
+  const contentResponse = await axios.get(documentResponse.data.url)
+  const rawData = contentResponse.data
+
+  // optionally parse the results
+  if (parse) {
+    return tabDelimitedToArray(rawData)
+  }
+  return rawData
+}
+
+/**
+ * Request, wait and parse results of an Amazon Report
+ */
+export async function getReportHelper(
+  reportsClient: ReportsApiClientV20210630,
+  reportType: string,
+  reportParameters: ReportParameters,
+  sleepTime = 10,
+  maxAttempts = 15,
+  parse = true,
```

Comment:
These should all be in an options object. It will not be clear at all when you call this what number goes to what without having to look up the fn signature every time.

Reply:
changed to object